### PR TITLE
Align SpawnProtocol with style guide

### PR DESCRIPTION
**Pull request by johanandren**
SpawnProtocol had some old ideas, now updated with new ideas.

Refs #27318

### CODE REVIEW

**Comment by johanandren** [comment]
To detail the changes:

* `apply` and `create` as factory methods for behaviors
* top level message type in `TheActor.Command`
* No fancy curry apply for ask

File: akka-actor-typed/src/main/scala/akka/actor/typed/SpawnProtocol.scala
Comment (raboof) [inline akka-actor-typed/src/main/scala/akka/actor/typed/SpawnProtocol.scala]: I know you just moved it, but I think we should update this documentation, as it's no longer accurate.

**Review by raboof — COMMENTED** [review]
Docs need an update, LGTM after that

File: akka-actor-typed/src/main/scala/akka/actor/typed/SpawnProtocol.scala
Comment (johanandren) [reply akka-actor-typed/src/main/scala/akka/actor/typed/SpawnProtocol.scala]: You mean the or-else part? I think the rest is correct (and the stopping part I just added since I saw questions on discuss about that).

Comment (raboof) [reply akka-actor-typed/src/main/scala/akka/actor/typed/SpawnProtocol.scala]: yeah `orElse` and it mentions `.behavior` rather than `.apply` still

Comment (johanandren) [reply akka-actor-typed/src/main/scala/akka/actor/typed/SpawnProtocol.scala]: Fixed!

**Review by raboof — APPROVED** [review]

**Review by jrudolph — APPROVED** [review]
LGTM

**Pull request merged**
